tests/test_writer.py

```diff
@@ -0,0 +1,59 @@
+import os
+import fcntl
+from multiprocessing import Process, Queue
+from datetime import datetime
+import pytest
+from warcprox.mitmproxy import ProxyingRecorder
+from warcprox.warcproxy import RecordedUrl
+from warcprox.writer import WarcWriter
+from warcprox import Options
+
+recorder = ProxyingRecorder(None, None, 'sha1', url='http://example.com')
+
+recorded_url = RecordedUrl(url='http://example.com', content_type='text/plain',
+                           status=200, client_ip='5.5.5.5',
+                           request_data=b'abc',
+                           response_recorder=recorder,
+                           remote_ip='6.6.6.6',
+                           timestamp=datetime.utcnow())
+
+
+def lock_file(queue, filename):
+    """Try to lock file and return 1 if successful, else return 0.
+    It is necessary to run this method in a different process to test locking.
+    """
+    try:
+        fi = open(filename, 'ab')
+        fcntl.lockf(fi, fcntl.LOCK_EX | fcntl.LOCK_NB)
+        fi.close()
+        queue.put('1')
+    except IOError:
+        queue.put('0')
+
+
+@pytest.mark.parametrize("no_warc_open_suffix,lock_result", [
+                         (True, '0'),
+                         (False, '1')])
+def test_warc_writer_locking(tmpdir, no_warc_open_suffix, lock_result):
```

Comment:
I wouldn't even bother testing the case of `no_warc_open_suffix=False`. No need to parameterize this test.

Also, test that unlocking works. I.e. try locking again in another subprocess after calling close_writer() and check that it succeeds.

Reply:
VERY good point, it succeeds indeed, I checked it. I'm updating the test.